py/desispec/workflow/desi_proc_funcs.py

```diff
@@ -715,6 +736,119 @@ def create_desi_proc_batch_script(night, exp, cameras, jobdesc, queue, runtime=N
+                                  gpuextract=None,
+                                  ):
+    """
+    Generate a SLURM batch script to be submitted to the slurm scheduler to run desi_proc.
+
+    Args:
+        night: str or int. The night the data was acquired.
+        tileid: str or int. The tile id for the data.
+        queue: str. Queue to be used.
+
+    Options:
+        runtime: str. Timeout wall clock time.
+        batch_opts: str. Other options to give to the slurm batch scheduler (written into the script).
+        system_name: name of batch system, e.g. cori-haswell, cori-knl.
+        mpistdstars: bool. Whether to use MPI for stdstar fitting.
+        gpuspecter: bool. Whether to use gpu_specter.
+        gpuextract: bool. Whether to perform gpu extraction with gpu_specter.
+
+    Returns:
+        scriptfile: the full path name for the script written.
+
+    """
+
+    batchdir = get_desi_proc_batch_file_path(night)
+
+    os.makedirs(batchdir, exist_ok=True)
+
+    jobname = get_desi_proc_tilenight_batch_file_name(night, tileid)
+
+    timingfile = f'{jobname}-timing-$SLURM_JOBID.json'
+    timingfile = os.path.join(batchdir, timingfile)
+
+    scriptfile = os.path.join(batchdir, jobname + '.slurm')
+
+    batch_config = batch.get_config(system_name)
+    threads_per_core = batch_config['threads_per_core']
+    gpus_per_node = batch_config['gpus_per_node']
+
+    nodes = 1
+    if system_name == 'perlmutter-gpu':
+        ncores = 64
+    elif system_name == 'perlmutter-cpu':
+        ncores = 128
+    else:
+        raise RuntimeError('tilenight batch jobs only supported on Perlmutter')
+
+    if runtime is None:
+        runtime = 30
+
+    runtime_hh = int(runtime // 60)
+    runtime_mm = int(runtime % 60)
+
+    with open(scriptfile, 'w') as fx:
+        fx.write('#!/bin/bash -l\n\n')
+        fx.write('#SBATCH -N {}\n'.format(nodes))
+        fx.write('#SBATCH --qos {}\n'.format(queue))
+        for opts in batch_config['batch_opts']:
+            fx.write('#SBATCH {}\n'.format(opts))
+        if batch_opts is not None:
+            fx.write('#SBATCH {}\n'.format(batch_opts))
+        if system_name == 'perlmutter-gpu':
+            # perlmutter-gpu requires projects name with "_g" appended
+            fx.write('#SBATCH --account desi_g\n')
+        else:
+            fx.write('#SBATCH --account desi\n')
+        fx.write('#SBATCH --job-name {}\n'.format(jobname))
+        fx.write('#SBATCH --output {}/{}-%j.log\n'.format(batchdir, jobname))
+        fx.write('#SBATCH --time={:02d}:{:02d}:00\n'.format(runtime_hh, runtime_mm))
+        fx.write('#SBATCH --exclusive\n')
+
+        fx.write('\n')
+
+        cmd = 'desi_proc_tilenight'
+        cmd += f' -n {night}'
+        cmd += f' -t {tileid}'
+        cmd += f' --timingfile {timingfile}'
+        if mpistdstars:
+            cmd += f' --mpistdstars'
+        if gpuspecter:
+            cmd += f' --gpuspecter'
+        if gpuextract:
+            cmd += f' --gpuextract'
```

Comment:
Let's make the default such that perlmutter-gpu implies using `--gpuspecter` and `--gpuextract`, and perlmutter-cpu implies not using them.  I definitely want that for `--gpuextract` (default to using GPU if you request a system with GPUs), which also implied `--gpuspecter` on perlmutter-gpu; I'm less certain about the best default for `--gpuspecter` on perlmutter-cpu since it can be meaningfully used with good speed benefits even on a pure-CPU system, and so maybe it should just always be the default.

Reply:
done. on perlmutter-cpu cpu_specter remains the default, unless `--gpuspecter` is specified on the command line.